draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,669 @@
+
+## 4. Definitions
+
+### 4.1 Interaction event
+
+Interaction event represent the user's current interaction purpose with Logto. Logto will use the interaction event to track the user's interaction progress and provide the corresponding interaction validation and actions.
+
+The following are the types of interaction event supported by Logto:
+
+- **Register**: A new user signs up for a account in Logto.
+- **SignIn**: An existing user signs in to their account in Logto.
+- **ForgotPassword**: A user forgets the password and initiates an account recovery process to reset password.
+
+### 4.2 Identifiers
+
+The following are the types of `identifiers` supported by Logto that can be used to uniquely identify a user in the interaction:
+
+- **Username**: A user's unique username.
+- **Email**: A verified user's email address.
+- **Phone**: A verified user's phone number.
+- **Social identity**: A verified user identity data provided by a social identity provider.
+- **SSO identity**: A verified user identity data provided by a enterprise SSO connector.
+
+### 4.3 Verification record
+
+A verification record contains the necessary security information to verify a user's identity. One or more verification records can be created and verified using the Experience APIs during the sign-in process.
+
+#### 4.3.1. Primary verification
+
+Primary verification can be used to verify the user's identity with a given identifier. The following are the primary verification record types supported by Logto:
+
+- **Password**: The user's password, which is the most common method for verifying identity. It can be used to verify the user's `username`, `email`, or `phone` identifier.
+- **Verification code**: A one-time code sent to the user's email or phone number for identifier verification. It can be used to verify the user's `email` or `phone` identifier.
+- **Social authentication**: An authenticated social identity used to directly identify and verify a user.
+- **SSO authentication**: An authenticated SSO identity used to directly identify and verify a user.
+
+#### 4.3.2. Multi-factor authentication(MFA) verification
+
+MFA is a security process that requires more than one verification factor to confirm a user's identity. It can be used as an additional security check during user interaction. Developers can enable MFA using Logto's sign-in experience settings. Once enabled, users will be prompted to provide additional MFA verification records during the interaction. The following are the MFA verification record types supported by Logto:
+
+- **TOTP code**: A time-based one-time code generated by a TOTP app for identity verification.
+- **WebAuthn**: A WebAuthn credential used to verify a user's identity.
+- **Backup code**: A set of backup codes used to verify a user's identity when other MFA verification methods are unavailable.
```

Comment:
Same as the passkey as a primary sign-in method. Prefer to update the RFC once the full feature spec is ready. As it may involve multiple flow and configuration updates.